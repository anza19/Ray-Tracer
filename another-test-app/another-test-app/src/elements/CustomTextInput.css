.custom-text-input {
    width: 300px;
    padding: 28px;
    margin: 28px;
    font: bold 1.8em "Consolas", serif;
  }