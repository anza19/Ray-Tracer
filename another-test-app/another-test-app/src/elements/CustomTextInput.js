import React from 'react';
import PropTypes from 'prop-types';

import './CustomTextInput.css';

const CustomTextInput = ({textInputValue}) => <input type="text" className="custom-text-input" name="action-txt" readOnly value={textInputValue} />

CustomTextInput.propTypes = {
    textInputValue: PropTypes.string.isRequired,
};

export default CustomTextInput;