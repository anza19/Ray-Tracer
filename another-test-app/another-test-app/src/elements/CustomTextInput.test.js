import CustomTextInput from "./CustomTextInput";

describe("CustomTextInput component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <CustomTextInput textInputValue={'test'} />
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  
  it("should display proper text value", () => {
    const wrapper = shallow(
      <CustomTextInput textInputValue={'test'} />
    );
  
    expect((wrapper).find('input').props().value).toEqual('test');
  });
});