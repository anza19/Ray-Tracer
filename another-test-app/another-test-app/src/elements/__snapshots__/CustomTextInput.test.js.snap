// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput component renders correctly 1`] = `
<input
  className="custom-text-input"
  name="action-txt"
  readOnly={true}
  type="text"
  value="test"
/>
`;
